src/Tasks/Microsoft.NET.Build.Tasks/PreprocessPackageDependenciesDesignTime.cs

```diff
@@ -97,11 +97,7 @@ protected override void ExecuteCore()
             DependenciesDesignTime = DependenciesWorld.Select(itemKvp =>
             {
                 var newTaskItem = new TaskItem(itemKvp.Key);
-                foreach (var metadataKvp in itemKvp.Value.ToDictionary())
```

Comment:
nit: Did you do a profile on this? What is the improvement?

In general I don't like perf change without tests to lock the perf improvement to prevent future regression or at least some profile it to show visible changes. This change is ok considering it is equality readable.

Reply:
This was mostly about making the code clearer. The reduction in allocation is a bonus.

I'm interested to know how you lock perf improvements in with tests.